### ISSUE TITLE
Implement more secrets checks

### ISSUE THREAD

**Issue by Fennerr** [toniblyx/prowler#3085]
### New feature motivation

Similar to the secrets checks for the other services (lambda/ec2/ecs/etc), more checks can be implemented

### Solution Proposed

Elastic Beanstalk:
* Configuration files (.ebextensions) used for environment customization might contain secrets.
* aws elasticbeanstalk describe-environment-resources --environment-name [Environment_Name]

API Gateway:
* Integration request parameters or mapping templates can potentially include secrets.
* aws apigateway get-integration --rest-api-id [RestApi_Id] --resource-id [Resource_Id] --http-method [HTTP_Method]

CodeBuild and CodePipeline:
* Buildspec files or pipeline definitions could contain embedded secrets, especially in environment variable definitions.
* aws codebuild batch-get-projects --names [Project_Name]
* aws codepipeline get-pipeline --name [Pipeline_Name]

Glue Jobs:
* ETL scripts used in AWS Glue can include hard-coded credentials for data sources or sinks.
* Check the script located at ScriptLocation and arguements passed using DefaultArguments
* aws glue get-job --job-name [Job_Name] 
* aws glue get-connection --name [Connection_Name]

Step Functions:
* State machine definitions might include sensitive information in task parameters. 
* aws stepfunctions describe-state-machine --state-machine-arn [State_Machine_Arn]

AppSync:
* Resolver mapping templates or data source configurations can include hard-coded API keys or database credentials.
* aws appsync get-graphql-api --api-id [API_Id]

This might not be all the API calls that need to be made to get the secrets, but should be a good starting point

### Describe alternatives you've considered

None

**Comment by Fennerr** [comment]
More secrets to consider:

AWS CodeCommit:
* Check for secrets in AWS CodeCommit repositories (including branches and commit history - a tool designed for git would be useful like trufflehog)

AWS Elastic Container Registry (ECR):
* Images stored in ECR might contain secrets. Scanning Dockerfiles and image layers for secrets can be crucial. (Trivy could be good here)

AWS Lambda Layer Contents:
* Lambda Layers are used to manage code and dependencies. Scanning the contents of these layers for secrets is as important as scanning the Lambda function code.

AWS Systems Manager State Manager:
* State Manager documents might include scripts or commands that contain secrets.

AWS Batch Job Definitions:
* Similar to ECS tasks, AWS Batch job definitions might include environment variables or command parameters that contain secrets.

Amazon SageMaker:
* Notebooks and model training scripts in SageMaker can sometimes include embedded secrets.

AWS Amplify Console:
* Check for secrets in Amplify app build settings and environment variables. (should be in environment secrets)

AWS Data Pipeline:
* Data Pipeline definitions, especially the ones that contain custom scripts or SQL commands, could have embedded secrets.

AWS Glue Data Catalog:
* Scanning AWS Glue Data Catalog for database connection details that might include hardcoded credentials.

**Comment by n4ch04** [comment]
Hi @Fennerr, all of them interesting ideas, but maybe we need to rethink a little bit our "secrets detection engine" ...
The `detect-secrets` package generates false positives and the current way of scanning generates a lot of resource exhaustion ...

**Comment by Fennerr** [comment]
This is true. Maybe move it to an optional flag and/or a config option to select what secrets you want to scan for

It would also be nice to generalize the way that secrets are scanned for so that stuff like writing to the temp files to disk, search for secrets, and using multiprocessing for this (as it's cpu intensive) can just be handled in one place.

**Comment by Gfuen** [comment]
@Fennerr Was there any decision on this ticket going forward? Because if the detect-secrets package is going to be changed I would love to work on that. Thanks!

**Comment by jfagoagas** [comment]
Hello @Gfuen there are no plans on changing the `detect-secrets` package. Do you have any suggestion for a different Python package to look for secrets?

Related to the checks requested above they are still pending to be done.

**Comment by jfagoagas** [comment]
For anyone looking to contribute to this issues take a look to our [DevGuide](https://docs.prowler.com/developer-guide/introduction).

This check is an example for how to check secrets in AWS Lambda functions code:
[https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/service[…]ion_no_secrets_in_code/awslambda_function_no_secrets_in_code.py](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/services/awslambda/awslambda_function_no_secrets_in_code/awslambda_function_no_secrets_in_code.py)

**Comment by Jeevan-88** [comment]
Hi! I'd love to work on this issue as my first open source contribution. I have AWS certifications (Cloud Practitioner, Solutions Architect, Developer Associate) and hands-on experience with AWS security projects. I'd like to start by implementing the CodeBuild secrets check. Can I be assigned?

**Comment by andoniaf** [comment]
Sure @Jeevan-88 ! Go ahead, let us know if you need any help 👍 you can also join our Prowler Community Slack at [goto.prowler.com/slack](https://goto.prowler.com/slack)

**Comment by danibarranqueroo** [comment]
Re-opened the issue because there are more checks ideas here and maybe someone still wants to contribute 🚀

**Comment by chirag1206** [comment]
Hi, I'd like to work on this. I've set up the dev environment, reviewed the existing secretsmanager service and all five current checks, and confirmed all 99 tests pass locally.
For a first contribution I was thinking of implementing secretsmanager_secret_missing_tags — flagging secrets with no tags at all, since untagged secrets can't be filtered by owner or environment for access governance. The tags field is already available in the service model so no new API calls are needed.
Does this fit the scope of what you had in mind for this issue? Happy to adjust the approach before I start coding.

**Comment by chirag1206** [comment]
Update: After reading the issue more carefully, I've changed direction. I've implemented apigateway_restapi_no_secrets_in_stage_variables instead — scanning API Gateway REST API stage variables for hardcoded secrets using the detect-secrets library, following the same pattern as the recently merged Step Functions check (#10570).

Changes:
- Extended Stage model in apigateway_service.py to capture stage variables
- Added new check apigateway_restapi_no_secrets_in_stage_variables
- 4 tests passing, full apigateway suite (43 tests) passing

PR coming shortly.

**Comment by danibarranqueroo** [comment]
I've created individual issues for each one of the secrets checks ideas that were here. With that, I'm closing this issue! 🙌
